src/explorer/DeploymentSlotsNode.ts

```diff
@@ -60,4 +61,60 @@ export class DeploymentSlotsNode extends NodeBase {
     private get azureAccount(): AzureAccountWrapper {
         return this.getTreeDataProvider<AppServiceDataProvider>().azureAccount;
     }
-}
\ No newline at end of file
+
+    public async createNewDeploymentSlot(): Promise<void> {
+        let slotName = '';
+        let slotNameOkay = false;
+
+        // a webapp cannot have more than 4 slots (excluding production)
+        const slotNodes = await this.getChildren();
+        const slotLabels = slotNodes.map(node => {
+            return node.label;
+        });
+
+        while (!slotNameOkay) {
+            slotName = await window.showInputBox({
+                prompt: 'Enter a unique name for the new deployment slot',
+                validateInput: (value: string) => {
+                    value = value ? value.trim() : '';
+                    if (!value.match(/^[a-z0-9\-]{1,60}$/ig)) {
+                        return 'Name should be 1-60 characters long and can only include alphanumeric characters and hyphens.';
+                    }
+
+                    // Can not have identical slot names OR production
+                    if (!this.checkSlotNameAvailability(value, slotLabels)) {

```

Comment:
Use the WebsiteManagementClient.checkNameAvailability() method for this job.

Reply:
It's up to Nathan - but I actually prefer the existing code because it can be done synchronously inside the inputbox validate function